### PR TITLE
ENG-13401 clean up  http sessions after catalog update

### PR DESCRIPTION
**Pull request by zeemanhe**

### CODE REVIEW

File: src/frontend/org/voltdb/RealVoltDB.java [hunk 1]
```diff
@@ -3470,7 +3470,7 @@ public CatalogContext catalogUpdate(
                 // 3. update HTTPClientInterface (asynchronously)
                 // This purges cached connection state so that access with
                 // stale auth info is prevented.
-                if (m_adminListener != null)
+                if (m_adminListener != null && m_catalogContext.authSystem.isSecurityEnabled())
```
Comment (akhanzode) [inline src/frontend/org/voltdb/RealVoltDB.java]: You need to notify regardless of security enabled or not. If security was enabled and then flipped to disabled old session cookies will give you old authResult thus failing those calls.

File: src/frontend/org/voltdb/utils/HTTPAdminListener.java [hunk 2]
```diff
@@ -367,4 +360,13 @@ public void notifyOfCatalogUpdate() {
             m_log.error("Failed to update HTTP interface after catalog update", ex);
         }
     }
+
+    private static class HttpSessionIdManager extends HashSessionIdManager {
+        @Override
+        public void doStop() throws Exception {
+            synchronized (this) {
```
Comment (akhanzode) [inline src/frontend/org/voltdb/utils/HTTPAdminListener.java]: This does not block other jetty threads from getting old session information as they dont block. How does this class help?

File: src/frontend/org/voltdb/RealVoltDB.java (hunk 1 above)
Comment (zeemanhe) [reply src/frontend/org/voltdb/RealVoltDB.java]: Will revert back the change. Are there any use cases that the security is on and off?

File: src/frontend/org/voltdb/utils/HTTPAdminListener.java (hunk 2 above)
Comment (zeemanhe) [reply src/frontend/org/voltdb/utils/HTTPAdminListener.java]: Correct. It does not block getting old sessions.  The intention is to block  the session addition and removal from the session cache.  Should getting session be blocked too? The old sessions seem still valid.

**Pull request merged**